[Bug]: Failed to load exercise progress data

### What happened?

I have the following integrations added:
IOS
Strava
Hevy

I synced all of my providers and now when I go to Reports > Exercise Progress, I get the following error: `Failed to load exercise progress data`

In developer tools, I get the error message:
```
[ERROR] Query Error:  ZodError: [
  {
    "expected": "string",
    "code": "invalid_type",
    "path": [
      0,
      "provider_name"
    ],
    "message": "Invalid input: expected string, received null"
  },
  {
    "expected": "string",
    "code": "invalid_type",
    "path": [
      1,
      "provider_name"
    ],
    "message": "Invalid input: expected string, received null"
  }
]
    at T (exerciseEntryService-DEtLz3fU.js:1:4028)
```

### Steps to reproduce

1. Import Strava, Hevy, and IOS data
2. Go to Reports >  Exercise Progress
3. Receive Error

### Expected behavior

I expect for all of my exercise data to show up in the reports page.

### How are you running SparkyFitness?

Docker Compose

### SparkyFitness version

v0.16.5.7

### Which area is affected?

Web frontend

### Relevant logs or screenshots

```shell
docker compose logs:
sparkyfitness-frontend-1  | 10.1.20.1 - - [16/Apr/2026:03:12:30 +0000] "GET /api/auth/get-session HTTP/1.1" 200 444 "https://xxxxxx.com/reports" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/146.0.0.0 Safari/537.36"
sparkyfitness-frontend-1  | 10.1.20.1 - - [16/Apr/2026:03:12:30 +0000] "GET /api/auth/get-session HTTP/1.1" 200 444 "https://xxxxxx.com/reports" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/146.0.0.0 Safari/537.36"
sparkyfitness-frontend-1  | 10.1.20.1 - - [16/Apr/2026:03:12:30 +0000] "GET /api/exercise-entries/progress/e07fda60-2693-4291-8bf7-925eb11aea78?startDate=2026-04-01&endDate=2026-04-15&aggregationLevel=daily HTTP/1.1" 304 0 "https://xxxxxxxx.com/reports" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/146.0.0.0 Safari/537.36"
sparkyfitness-frontend-1  | 10.1.20.1 - - [16/Apr/2026:03:12:30 +0000] "GET / HTTP/1.0" 200 2666 "-" "-"
sparkyfitness-frontend-1  | 10.1.20.1 - - [16/Apr/2026:03:12:31 +0000] "GET /api/exercise-entries/progress/e07fda60-2693-4291-8bf7-925eb11aea78?startDate=2026-04-01&endDate=2026-04-15&aggregationLevel=daily HTTP/1.1" 304 0 "https://xxxxxxx.com/reports" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/146.0.0.0 Safari/537.36"
```

### Anything else?

Other than docker compose logs and the developer tools, I am having issue finding any other logs that might help troubleshoot this. Any advice on where to find logs that are needed would be helpful.

Thank you,

--- a/SparkyFitnessFrontend/src/components/MealBuilder.tsx
+++ b/SparkyFitnessFrontend/src/components/MealBuilder.tsx
@@ -36,6 +36,7 @@ import {
   useCreateFoodEntryMealMutation,
   useUpdateFoodEntryMealMutation,
 } from '@/hooks/Diary/useFoodEntries';
+import { Textarea } from './ui/textarea';
 
 interface MealBuilderProps {
   mealId?: string; // Optional: if editing an existing meal template
@@ -571,7 +572,7 @@ const MealBuilder: React.FC<MealBuilderProps> = ({
         <Label htmlFor="mealDescription">
           {t('mealBuilder.mealDescription', 'Description (Optional)')}
         </Label>
-        <Input
+        <Textarea
           id="mealDescription"
           value={mealDescription}
           onChange={(e) => setMealDescription(e.target.value)}

--- a/SparkyFitnessFrontend/src/components/MealBuilder.tsx
+++ b/SparkyFitnessFrontend/src/components/MealBuilder.tsx
@@ -36,7 +36,7 @@ import {
   useCreateFoodEntryMealMutation,
   useUpdateFoodEntryMealMutation,
 } from '@/hooks/Diary/useFoodEntries';
-import { Textarea } from './ui/textarea';
+import { Textarea } from '@/components/ui/textarea';
 
 interface MealBuilderProps {
   mealId?: string; // Optional: if editing an existing meal template

--- a/shared/src/schemas/api/ExerciseEntries.api.zod.ts
+++ b/shared/src/schemas/api/ExerciseEntries.api.zod.ts
@@ -213,7 +213,7 @@ export const exerciseProgressResponseSchema = z.object({
   image_url: z.string().nullable(),
   distance: z.number().nullable(),
   avg_heart_rate: z.number().nullable(),
-  provider_name: z.string(),
+  provider_name: z.string().nullable(),
   sets: z.array(exerciseEntrySetRequestSchema),
 });
 

